src/main/kotlin/com/buildit/bookit/v1/user/dto/User.kt

```diff
@@ -0,0 +1,31 @@
+package com.buildit.bookit.v1.user.dto
+
+import com.buildit.bookit.auth.UserPrincipal
+import com.buildit.bookit.v1.booking.dto.Booking
+import com.fasterxml.jackson.annotation.JsonIgnore
+import org.hibernate.annotations.GenericGenerator
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.GeneratedValue
+import javax.persistence.Id
+
+@Entity
+data class User(
+    @Column(unique = true, nullable = false)
+    val externalId: String,
+    @Column(nullable = false) @JsonIgnore
+    val givenName: String,
+    @Column(nullable = false) @JsonIgnore
+    val familyName: String,
+    @Id @GeneratedValue(generator = "uuid2") @GenericGenerator(name = "uuid2", strategy = "uuid2")
+    val id: String? = null
+) {
+    val name get() = "$givenName $familyName"
```

Comment:
sort of a hack to reuse the User entity as the User response but it still seems "simple enough"...